### PR TITLE
feat: add `pox-3` to contract list in `/v2/pox` RPC response

### PR DESCRIPTION
**Pull request by zone117x**
Adds a new `pox-3` entry to the `contract_versions` array returned by the `/v2/pox` RPC endpoint. Sample output from regtest/krypton node:

```json
{
  "contract_id": "ST000000000000000000002AMW42H.pox-2",
  "pox_activation_threshold_ustx": 600057388429055,
  "first_burnchain_block_height": 0,
  "current_burnchain_block_height": 111,
  "prepare_phase_block_length": 1,
  "reward_phase_block_length": 4,
  "reward_slots": 8,
  "rejection_fraction": 3333333333333333,
  "total_liquid_supply_ustx": 60005738842905576,
  "current_cycle": {
    "id": 22,
    "min_threshold_ustx": 1875180000000000,
    "stacked_ustx": 0,
    "is_pox_active": false
  },
  "next_cycle": {
    "id": 23,
    "min_threshold_ustx": 1875180000000000,
    "min_increment_ustx": 7500717355363,
    "stacked_ustx": 0,
    "prepare_phase_start_block_height": 114,
    "blocks_until_prepare_phase": 3,
    "reward_phase_start_block_height": 115,
    "blocks_until_reward_phase": 4,
    "ustx_until_pox_rejection": 8484139029839119000
  },
  "min_amount_ustx": 1875180000000000,
  "prepare_cycle_length": 1,
  "reward_cycle_id": 22,
  "reward_cycle_length": 5,
  "rejection_votes_left_required": 8484139029839119000,
  "next_reward_cycle_in": 4,
  "contract_versions": [
    {
      "contract_id": "ST000000000000000000002AMW42H.pox",
      "activation_burnchain_block_height": 0,
      "first_reward_cycle_id": 0
    },
    {
      "contract_id": "ST000000000000000000002AMW42H.pox-2",
      "activation_burnchain_block_height": 107,
      "first_reward_cycle_id": 22
    },
    {
      "contract_id": "ST000000000000000000002AMW42H.pox-3",
      "activation_burnchain_block_height": 3000000,
      "first_reward_cycle_id": 600000
    }
  ]
}
```

TODO: 
* [ ] Tests

### CODE REVIEW

File: src/net/rpc.rs [hunk 1]
```diff
@@ -317,6 +317,12 @@ impl RPCPoxInfoData {
             ))?
             + 1;
 
+        let pox_3_first_cycle = burnchain
+            .block_height_to_reward_cycle(burnchain.pox_constants.pox_3_activation_height as u64)
+            .ok_or(net_error::ChainstateError(
+                "PoX-3 first reward cycle begins before first burn block height".to_string(),
+            ))?;
+
```
Comment (zone117x) [inline src/net/rpc.rs:325]: Is this accurate? The `pox_2_first_cycle` calculation above adds a ` + 1` -- is that also needed here?

Comment (zone117x) [reply src/net/rpc.rs:325]: Based off what I'm seeing after allowing epoch2.4 configuration in https://github.com/stacks-network/stacks-blockchain/pull/3705, it seems like this does need a `+ 1`. Not sure though, need feedback from someone more familiar.

Comment (obycode) [reply src/net/rpc.rs:325]: Yes, I think this should be +1. I believe the pox-3 contract gets deployed once the activation burn height is reached, so it actually ends up in the next block.

Comment (zone117x) [reply src/net/rpc.rs:326]: Thanks @obycode! Fixed in https://github.com/stacks-network/stacks-blockchain/pull/3704/commits/5e6c36ed8076ae5ecf071e2c9a087d18fe8ea517

**Review by pavitthrap — APPROVED** [review]

**Review by kantai — APPROVED** [review]

**Pull request merged**